packages/devtools_app/lib/src/screens/memory/primitives/class_name.dart

```diff
@@ -0,0 +1,285 @@
+// Copyright 2022 The Chromium Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'package:flutter/foundation.dart';
+import 'package:vm_service/vm_service.dart';
+
+import '../../../primitives/simple_items.dart';
+
+@immutable
+class HeapClassName {
+  HeapClassName({required this.className, required library})
+      : library = _normalizeLibrary(library) {
+    assert(
+      !isCore || !isDartOrFlutter,
+      'isCore and isDartOrFlutter must be exclusive',
+    );
+  }
+
+  HeapClassName.fromClassRef(ClassRef? classRef)
+      : this(
+          library: _library(
+            classRef?.library?.name,
+            classRef?.library?.uri,
+          ),
+          className: classRef?.name ?? '',
+        );
+
+  HeapClassName.fromHeapSnapshotClass(HeapSnapshotClass? theClass)
+      : this(
+          library: _library(
+            theClass?.libraryName,
+            theClass?.libraryUri.toString(),
+          ),
+          className: theClass?.name ?? '',
+        );
+
+  static String _library(String? libName, String? libUrl) {
+    libName ??= '';
+    if (libName.isNotEmpty) return libName;
+    return libUrl ?? '';
```

Comment:
this could be a one liner
`return libName ?? (libUrl ?? '');`

Reply:
If libName is empty, I want to return libUrl
Reduced to two lines